portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -147,17 +147,10 @@ private String getUploadedFileUrl(HttpServletRequest request,
                                     String formInputElementName) {
     BlobstoreService blobstoreService =
         BlobstoreServiceFactory.getBlobstoreService();
-    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
-    List<BlobKey> blobKeys = blobs.get("image");
 
-    // User submitted form without selecting a file, so we can't get a URL. (dev
-    // server)
-    if (blobKeys == null || blobKeys.isEmpty()) {
-      return null;
-    }
 
-    // Our form only contains a single file input, so get the first index.
-    BlobKey blobKey = blobKeys.get(0);
+    // Get the value of the blob key from our upload.
+    BlobKey blobKey = getBlobKey(request, formInputElementName);
```

Comment:
Looks like this function isn't used anymore.

Reply:
Should I delete this function though, I wasn't sure about this as it's a more efficient implementation, it just doesn't work on the specific platform the server is deployed on. I was thinking I should leave it in case the user deploys on a different platform then they just need to change two lines of code? I was really uncertain though.

Reply:
What are your thoughts on this?

Reply:
I prefer not to include code that I'm not committed to supporting. Today, this function works if you change two lines of code. But as the code grows and changes, it will naturally get out of sync since nothing is calling it (especially since we don't have tests). By the time someone needs it, it's probably not a drop-in change anymore, and maybe not useful at all.

If someone _does_ need it today, then it's worth supporting it with tests, having it configurable without a change to the source, etc. But a lot of the time, what you think will be useful down the line ends up not materializing the way you expected.

http://go/tott/438

Reply:
Hmm ok, I'll keep that in mind. No permission on the go link unfortunately, what does it mention?

Reply:
I deleted this function.

Reply:
Oh no, I've been posting these links in lots of comments. Whoops. Can you see this version? https://drive.google.com/file/d/0B-yWrS8S9kyqQU1VbDF5N3hndW8/view

If not, it's some recommendations based on the YAGNI principle: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

Reply:
No, I requested access though. Thanks for the other link.